lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php

```diff
@@ -53,6 +53,14 @@ public function connect(array $params, $username = null, $password = null, array
                 $pdo->setAttribute(PDO::PGSQL_ATTR_DISABLE_PREPARES, true);
             }
 
+            /* defining client_encoding via SET NAMES to avoid inconsistent DSN support
+             * - the 'client_encoding' connection param only works with postgres >= 9.1
+             * - passing client_encoding via the 'options' param breaks pgbouncer support
+             */
+            if (isset($params['charset'])) {
+              $pdo->query('SET NAMES \''.$params['charset'].'\'');
```

Comment:
@deeky666 do you get what I am saying about PGSQL_LIBPQ_VERSION?

Please let me know if I am confused -- but that only tells us what CLIENT is being used -- NOT what server is being connected to.

The point is -- just because you are connecting from a libpq client that supports client_encoding -- that doesn't mean it's going to work. The server has to suppport it of course. The client_encoding option wasn't added until 9.1 (both client and server).

If we make the logic based on the client version it means two things:

1) we introduce a bug where if you're connecting to, say an 8.4 server from a 9.1 client -- we'll end up passing client_encoding (because PGSQL_LIBPQ_VERSION tells us 9.1) -- but the 8.4 server is going to reject it.

Can somebody else chime in here? I want to make sure I am not the one understanding this wrong -- but I am pretty sure I am understanding this properly.

@riccardonar you're using 9.0 client anyway -- so regardless of what we did you're going to see the same behavior (IE, SET NAMES 'xxx' is the only way to set the character encoding reliably)